src/macros.rs

```diff
@@ -1219,7 +1219,14 @@ impl MacroBranch {
         }
 
         // 5 = " => {"
-        let mut result = format_macro_args(context, self.args.clone(), shape.sub_width(5)?)?;
+        let old_body = context.snippet(self.body).trim();
+        let prefix_width = 5 + if context.config.format_macro_bodies() {
+            old_body.find(|ch| ch != '{').unwrap_or(old_body.len()) // " => {{..."
+        } else {
+            0
+        };
```

Comment:
If we hoist the declaration for `has_block_body` up from line 1248 could this be simplified as follows:

```rust
        let old_body = context.snippet(self.body).trim();
        let has_block_body = old_body.starts_with('{');
        let prefix_width = if has_block_body {
            // 6 = " => {{"
            6
        } else {
            // 5 = " => {"
            5
        };
```

Reply:
Changed, but kept the `if context.config.format_macro_bodies()`:
```rust
let prefix_width = if has_block_body && context.config.format_macro_bodies() {
    // 6 = " => {{"
    6
} else {
    // 5 = " => {"
    5
};
```

Reply:
Does the bug still pop up if `format_macro_bodies=false`?

I'm not sure we should conditionally set the prefix width based on the `format_macro_bodies` config argument since setting the prefix width to 5 when there are two opening braces is incorrect, and it's what lead to the bug in the first place.

Reply:
Yes, the bug still pop if `format_macro_bodies=false`.  From you explanation to the other comment about the macro body, I now better understand the intention, so I **removed** the `if context.config.format_macro_bodies()` as you suggested.

Reply:
Great! Apologies if it wasn't totally clear what I was trying to accomplish with the refactored code snippet above.